apps/web/src/api/hooks/useUploadToStorage.ts

```diff
@@ -0,0 +1,42 @@
+import axios from 'axios';
+import { useMutation, UseMutationOptions } from '@tanstack/react-query';
+import { IResponseError, MIME_TYPE_TO_FILE_EXTENSION, UploadTypesEnum } from '@novu/shared';
+
+import { useGetSignedUrl } from './useGetSignedUrl';
+
+export const useUploadToStorage = (
+  options: UseMutationOptions<string, IResponseError, { file: File; type: UploadTypesEnum }> = {}
+) => {
+  const { getSignedUrl } = useGetSignedUrl();
+  const { mutateAsync: uploadToStorage, ...mutationData } = useMutation(
+    async ({ file, type }) => {
+      const { signedUrl, path, additionalHeaders } = await getSignedUrl({
+        extension: MIME_TYPE_TO_FILE_EXTENSION[file.type],
+        type,
+      });
+      await axios.put(signedUrl, file, {
+        headers: {
+          'Content-Type': file.type,
+          ...additionalHeaders,
+        },
+        transformRequest: [
+          (data, headers) => {
+            if (headers) {
+              // eslint-disable-next-line
+              delete headers.Authorization;
+            }
+
+            return data;
+          },
+        ],
+      });
+
+      return path;
```

Comment:
the logic extracted from the branding page and moved to this reusable hook

Reply:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do not leave false praise (which can actually be damaging). Do look for something to sincerely praise.  -->
**praise:** Appreciate this reusable hook 🙏🏼 ‏